Testscripts/Windows/PERF-NETWORK-TCP-LATENCY-LAGSCOPE.ps1

```diff
@@ -118,33 +119,52 @@ collect_VM_properties
-            $resultMap = @{}
-            if ($properties) {
-                $resultMap["GuestDistro"] = $properties.GuestDistro
-                $resultMap["HostOS"] = $properties.HostOS
-                $resultMap["KernelVersion"] = $properties.KernelVersion
-            }
-            $resultMap["HostType"] = "Azure"
-            $resultMap["HostBy"] = $CurrentTestData.SetupConfig.TestLocation
-            $resultMap["GuestOSType"] = "Linux"
-            $resultMap["GuestSize"] = $clientVMData.InstanceSize
-            $resultMap["IPVersion"] = "IPv4"
-            $resultMap["ProtocolType"] = "TCP"
-            $resultMap["TestCaseName"] = $global:GlobalConfig.Global.$TestPlatform.ResultsDatabase.testTag
-            $resultMap["TestDate"] = $(Get-Date -Format yyyy-MM-dd)
-            $resultMap["LISVersion"] = "Inbuilt"
+            $testDate = $(Get-Date -Format yyyy-MM-dd)
             if ($currentTestData.SetupConfig.Networking -imatch "SRIOV") {
-                $resultMap["DataPath"] = "SRIOV"
+                $dataPath = "SRIOV"
             } else {
-                $resultMap["DataPath"] = "Synthetic"
+                $dataPath = "Synthetic"
+            }
+
+            $histogramFlag = $false
+            foreach ($line in $lagscopeReportLog) {
+                if ($line -imatch "Interval\(usec\)") {
+                    $histogramFlag = $true
+                    continue;
+                }
+                if ($histogramFlag -eq $false) {
+                    continue;
+                }
+                $interval = ($line.Trim() -replace '\s+',' ').Split(" ")[0]
+                $frequency = ($line.Trim() -replace '\s+',' ').Split(" ")[1]
+                if (($interval -match "^\d+$") -and ($frequency -match "^\d+$") -and ($interval -ne "0")) {
+                    $resultMap = @{}
+                    if ($properties) {
+                        $resultMap["GuestDistro"] = $properties.GuestDistro
+                        $resultMap["HostOS"] = $properties.HostOS
+                        $resultMap["KernelVersion"] = $properties.KernelVersion
+                    }
+                    $resultMap["HostType"] = "Azure"
+                    $resultMap["HostBy"] = $CurrentTestData.SetupConfig.TestLocation
+                    $resultMap["GuestOSType"] = "Linux"
+                    $resultMap["GuestSize"] = $clientVMData.InstanceSize
+                    $resultMap["IPVersion"] = "IPv4"
+                    $resultMap["ProtocolType"] = "TCP"
+                    $resultMap["TestCaseName"] = $global:GlobalConfig.Global.$TestPlatform.ResultsDatabase.testTag
+                    $resultMap["TestDate"] = $testDate
+                    $resultMap["LISVersion"] = "Inbuilt"
+                    $resultMap["DataPath"] = $dataPath
+                    $resultMap["MaxLatency_us"] = $maximumLat
+                    $resultMap["AverageLatency_us"] = $averageLat
+                    $resultMap["MinLatency_us"] = $minimumLat
```

Comment:
these values are treated as string, I previously met problem when inserting to database. Can we explicitly convert them to decimal?
$resultMap["MaxLatency_us"] = [Decimal]$maximumLat
$resultMap["AverageLatency_us"] = [Decimal]$averageLat
$resultMap["MinLatency_us"] = [Decimal]$minimumLat

Reply:
same as other numerical types